pkg/resources/pullSecret.go

```diff
@@ -16,40 +18,38 @@ const (
 	DefaultOriginPullSecretNamespace = "openshift"
 )
 
-// Gets the default pull secret for pulling container images from registry
-func GetDefaultPullSecret(context context.Context, client k8sclient.Client, inst *integreatlyv1alpha1.Installation) (corev1.Secret, error) {
+// Copys the default pull secret to a target namespace
+func CopyDefaultPullSecretToNameSpace(context context.Context, nameSpaceToCopy string, nameOfSecret string, inst *integreatlyv1alpha1.Installation, client k8sclient.Client) error {
 	if inst.Spec.PullSecret.Name == "" {
 		inst.Spec.PullSecret.Name = DefaultOriginPullSecretName
 	}
 	if inst.Spec.PullSecret.Namespace == "" {
 		inst.Spec.PullSecret.Namespace = DefaultOriginPullSecretNamespace
 	}
 
-	openshiftSecret := corev1.Secret{}
-
-	err := client.Get(context, types.NamespacedName{Name: inst.Spec.PullSecret.Name, Namespace: inst.Spec.PullSecret.Namespace}, &openshiftSecret)
-
-	return openshiftSecret, err
+	return CopySecret(context, client, inst.Spec.PullSecret.Name, inst.Spec.PullSecret.Namespace, nameOfSecret, nameSpaceToCopy)
 }
 
-// Copys the default pull secret to a target namespace
-func CopyDefaultPullSecretToNameSpace(context context.Context, nameSpaceToCopy string, nameOfSecret string, inst *integreatlyv1alpha1.Installation, client k8sclient.Client) error {
-	openshiftSecret, err := GetDefaultPullSecret(context, client, inst)
-
+func CopySecret(ctx context.Context, client k8sclient.Client, srcName, srcNamespace, destName, destNamespace string) error {
+	srcSecret := corev1.Secret{}
+	err := client.Get(ctx, types.NamespacedName{Name: srcName, Namespace: srcNamespace}, &srcSecret)
 	if err != nil {
 		return err
 	}
 
-	componentSecret := &corev1.Secret{
+	destSecret := &corev1.Secret{
 		Type: corev1.SecretTypeDockerConfigJson,
```

Comment:
I know it existed before your changes, but that's a strange secret type to specify :thinking: Is it necessary to specify a type here?

Reply:
yeah, this informs the cluster that this secret can be used for pulling in images to satisfy deployments / deploymentconfigs.